test_e2e.sh

```diff
@@ -1072,6 +1074,66 @@ function e2e::exechook_fail_retry() {
     fi
 }
 
+##############################################
+# Test exechook-success with GIT_SYNC_ONE_TIME
+##############################################
+function e2e::exechook_success_once() {
+    # First sync
+    echo "$FUNCNAME 1" > "$REPO"/file
+    git -C "$REPO" commit -qam "$FUNCNAME 1"
+
+    GIT_SYNC \
+        --period=100ms \
+        --one-time \
+        --repo="file://$REPO" \
+        --branch="$MAIN_BRANCH" \
+        --root="$ROOT" \
+        --link="link" \
+        --exechook-command="$EXECHOOK_COMMAND_SLEEPY" \
+        >> "$1" 2>&1 &
+
+    # Original test sleeps for 2 seconds, so sleep for 2 seconds more than
+    # hook's sleep time
+    sleep 5
+    assert_link_exists "$ROOT"/link
+    assert_file_exists "$ROOT"/link/file
+    assert_file_exists "$ROOT"/link/exechook
+    assert_file_exists "$ROOT"/link/link-exechook
+    assert_file_eq "$ROOT"/link/file "$FUNCNAME 1"
+    assert_file_eq "$ROOT"/link/exechook "$FUNCNAME 1"
+    assert_file_eq "$ROOT"/link/link-exechook "$FUNCNAME 1"
+}
+
+##############################################
+# Test exechook-fail with GIT_SYNC_ONE_TIME
+##############################################
+function e2e::exechook_fail_once() {
+    cat /dev/null > "$RUNLOG"
+        # First sync - return a failure to ensure that we try again
+        echo "$FUNCNAME 1" > "$REPO"/file
+        git -C "$REPO" commit -qam "$FUNCNAME 1"
+
+        GIT_SYNC \
+            --period=100ms \
+            --one-time \
+            --repo="file://$REPO" \
+            --branch="$MAIN_BRANCH" \
+            --root="$ROOT" \
+            --link="link" \
+            --exechook-command="$EXECHOOK_COMMAND_FAIL_SLEEPY" \
+            --exechook-backoff=1s \
+            >> "$1" 2>&1 &
+
+        # Check that exechook was called
+        # Original test sleeps for 2 seconds, so sleep for 2 seconds more than
+        # hook's sleep time
+        sleep 5
+        RUNS=$(cat "$RUNLOG" | wc -l)
```

Comment:
why would this be 2?

Reply:
You are right, bumped this down to 1. My apologies, made these tests before being able to test them (was having issues setting up Fedora VM). So just copied as much from similar tests as possible.

Reply:
Have it running now and ran the tests.